src/backends/onnx/onnx_utils.cc

```diff
@@ -274,8 +274,8 @@ CompareDimsSupported(
     const int max_batch_size)
 {
   // If the model configuration expects batching support in the model,
-  // then the model shape first dimension must be either -1 (unbounded)
-  // or value that larger than max_batch_size.
+  // then the model shape's first dimension must be either -1 (unbounded)
```

Comment:
Is this true. If the model shape's first dimension is not -1, then how can be know if it supports batching or not? @GuanLuo what does this mean?

Reply:
Here just saying that the batch dimension doesn't have to be -1. By the point where this function is called, the model config should be normalized. So if the model config says `max_batch_size : 8` and input dims is [16]. Then the actual input shape (retrieved from model) can be [-1, 16] or [9, 16] (as long as the value is >= 8)

Reply:
For TF we are more conservative in that we only allow TRTIS to assume the model understand batching (max_batch_size > 0) if the first dimension of every tensor is -1. It seems dangerous to assume that other tensors (without -1 batch dim) could support batching.

Reply:
I see your concern now. If the first dimension has to be -1, then people who doesn't know the detail of the model can change the `max_batch_size` without any concern. However, in the case here, they may find that the model suddenly can't be loaded after they change it to some larger value.

So I agree with you, I think being more conservative will be better.

Reply:
Updated the generated models and fixed this check to be more conservative (as discussed above).